06_modules_lab/03_bonus.py

```diff
@@ -1,13 +1,35 @@
-""" Write a program that searches current working directory
-for files larger than 1MB. Every time you find such a file print
-its name to the user.
-
-- When the program finds a large file. It should ask the user
-  a message asking if she wants to delete it, and delete the
-  file if requested
-
-- Take threshold and path as command line arguments
-
-Bonus: Use argparse module to parse command line arguments
-"""
-
+import os
+import argparse
+
+# Assign description to the help doc
+parser = argparse.ArgumentParser(description='Script search directory to find and delete large files.')
+
+# Add arguments
+parser.add_argument('-d', '--directory', type=str, help='Directory path', required=True)
+parser.add_argument('-s', '--size', type=int, help='minimum size to delete', required=True)
+
+# Array for all arguments passed to script
+args = parser.parse_args()
+
+# Assign args to variables
+search_dir = args.directory
+min_size_to_delete = args.size
+
+print "searching for files in '%s' bigger than %d bytes" % (search_dir, min_size_to_delete)
+
+if os.path.exists(search_dir):
+    for root, dirs, files in os.walk(search_dir):
+        for name in files:
+            #print "found file: %s/%s" % (root, name)
+
+            full_path = "{}\{}".format(root,name)

```

Comment:
Use `os.path.join(root, name)` to connect file path parts (it is more robust: In unix the path separator is /)